File: TeachersGuardAPI/Infraestructure/Mappers/ScheduleMapper.cs
using TeachersGuardAPI.App.DTOs.Schedule;
using TeachersGuardAPI.Domain.Entities;
using TeachersGuardAPI.Infraestructure.Models.Mongo;

namespace TeachersGuardAPI.Infraestructure.Mappers
{
    public class ScheduleMapper
    {
        public static Schedule MapScheduleDocumentToScheduleEntity(ScheduleDocument scheduleDocument) => new()
        {

            UserId = scheduleDocument.UserId,
            PlaceId = scheduleDocument.PlaceId,
            ScheduleId = scheduleDocument.Id.ToString(),
            End = scheduleDocument.EndTime,
            Start = scheduleDocument.StartTime,
            DayOfWeek = scheduleDocument.DayOfWeek,
        };

        public static ScheduleDto MapScheduleEntityToScheduleDto(Schedule schedule) => new()
        {
            DayOfWeek = schedule.DayOfWeek,
            End = schedule.End,
            PlaceId = schedule.PlaceId,
            ScheduleId = schedule.ScheduleId,
            Start = schedule.Start
        };
    }
}

File: lib/infraestructure/mappers/schedule_mapper.dart
import 'package:teachersguard/domain/entities/entities.dart';
import 'package:teachersguard/infraestructure/models/response/schedule/schedule_response.dart';

class ScheduleMapper {
  static Schedule mapScheduleResponseToScheduleEntity(
          ScheduleObjectResponse response) =>
      Schedule(
          id: response.scheduleId,
          place: Place(id: response.placeId, name: response.placeName),
          begin: response.start,
          end: response.end,
          dayOfWeek: response.dayOfWeek
      );
}
